dapp-oeth/src/components/buySell/SwapCurrencyPill.js

```diff
@@ -0,0 +1,881 @@
+            src={assetRootPath(`/images/currency/weth-icon-small.svg`)}
+          />
+        </div>
+      )}
+      <style jsx>{`
+        .coin-image {
+          width: 40px;
+          height: 40px;
+        }
+
+        .coin-image.small {
+          width: 24px;
+          height: 24px;
+        }
+
+        .mixed {
+          position: relative;
+        }
+
+        .coin-1 {
+          z-index: 1;
+        }
+
+        .coin-2 {
+          z-index: 2;
+          margin-left: -16px;
+        }
+
+        .coin-3 {
+          z-index: 3;
+          margin-left: -16px;
+        }
+
+        .coin-4 {
+          z-index: 4;
+          margin-left: -16px;
+        }
+      `}</style>
+    </div>
+  )
+}
+
+const coinToDisplay = {
+  eth: {
+    name: 'ETH',
+    symbol: 'ETH',
+  },
+  oeth: {
+    name: 'Origin Ether',
+    symbol: 'OETH',
+  },
+  weth: {
+    name: 'Wrapped Ether',
+    symbol: 'WETH',
+  },
+  reth: {
+    name: 'Rocket Pool ETH',
+    symbol: 'rETH',
+  },
+  steth: {
+    name: 'Liquid Staked Ether 2.0',
+    symbol: 'stETH',
+  },
+  frxeth: {
+    name: 'Frax Ether',
+    symbol: 'frxETH',
+  },
+  woeth: {
+    name: 'Wrapped Origin Ether',
+    symbol: 'WOETH',
+  },
+  sfrxeth: {
+    name: 'Staked Frax Ether',
+    symbol: 'sfrxETH',
+  },
+  mix: {
+    name: 'Redeem Mix',
+    symbol: 'frxEth, rETH, stETH, WETH',
+  },
+}
+
+const coinOrdering = [
+  'eth',
+  'weth',
+  'oeth',
+  'steth',
+  'reth',
+  'frxeth',
+  'sfrxeth',
+]
+
+const TokenSelectionModal = ({
+  tokens,
+  onClose,
+  onSelect,
+  conversion,
+  coinBalances,
+}) => {
+  const tokenData = orderBy(
+    tokens.map((token) => ({
+      token,
+      balance: parseFloat(coinBalances?.[token] || 0),
+      order: coinOrdering?.findIndex((current) => current === token) || 999,
+    })),
+    ['balance', 'order'],
+    ['desc', 'asc']
+  )
+
+  return (
+    <>
+      <div className="token-selection-modal">
+        <div className="content-backdrop" onClick={onClose} />
+        <div className="content-container">
+          {tokenData.map(({ token, balance }) => {
+            return (
+              <button
+                key={token}
+                className={`${
+                  token === 'oeth' ? 'oeth' : ''
+                }  d-flex justify-content-between align-items-center selectable`}
+                onClick={() => {
+                  onSelect(token)
+                  onClose()
+                }}
+              >
+                <div className="coin-view">
+                  <CoinImage coin={token} />
+                  <div className="coin-breakdown">
+                    <span className="name">{coinToDisplay?.[token]?.name}</span>
+                    <span className="coin mr-auto">
+                      {coinToDisplay?.[token]?.symbol}
+                    </span>
+                  </div>
+                </div>
+                <div className="balances">
+                  <span className="eth">
```

Comment:
nit: There seems to be no `.eth` class in the CSS below, should `.usd` class name be changed to `.usd`?